lib/main.ts

```diff
@@ -1,22 +1,45 @@
-/// <reference path="../typings/node/node.d.ts" />
+/// <reference path='../typings/node/node.d.ts' />
+/// <reference path='../typings/source-map/source-map.d.ts' />
 // Use HEAD version of typescript, installed by npm
-/// <reference path="../node_modules/typescript/bin/typescript.d.ts" />
+/// <reference path='../node_modules/typescript/bin/typescript.d.ts' />
 require('source-map-support').install();
-import ts = require("typescript");
-import fs = require("fs");
+import fs = require('fs');
+import SourceMap = require('source-map');
+import ts = require('typescript');
 
 export type ClassLike = ts.ClassDeclaration | ts.InterfaceDeclaration;
 
+export interface TranspilerOptions {
+  failFast?: boolean;
+  generateLibraryName?: boolean;
+  generateSourceMap?: boolean;
+}
+
 export class Transpiler {
+  // Position information.
+  generateSourceMap: boolean;
+  sourceMap: SourceMap.SourceMapGenerator;
+
+  generateLibraryName: boolean;
+  relativeFileName: string;
+  currentFile: ts.SourceFile;
+
   result: string = '';
+  column: number = 1;
+  line: number = 1;
+
   // Comments attach to all following AST nodes before the next 'physical' token. Track the earliest
   // offset to avoid printing comments multiple times.
   lastCommentIdx: number = -1;
-  currentFile: ts.SourceFile;
-  relativeFileName: string;
+
+  failFast: boolean;
   errors: string[] = [];
 
-  constructor(private failFast: boolean = false, private generateLibraryName: boolean = false) {}
+  constructor(opts: TranspilerOptions = {}) {
+    this.failFast = !!opts.failFast;
```

Comment:
No strong reason - I thought it was clearer to pull out the fields and coerce them into booleans here. I can change if you feel strongly about it.